server/datastore/mysql/migrations/tables/20250807141218_SoftwareIsKernelColumn.go

```diff
@@ -37,6 +37,18 @@ WHERE source IN ('rpm_packages', 'deb_packages')
 		return fmt.Errorf("failed to backfill software_titles.is_kernel column: %w", err)
 	}
 
+	if _, err := tx.Exec(`
+CREATE TABLE kernels (
+  id int unsigned NOT NULL AUTO_INCREMENT,
+  software_title_id int unsigned DEFAULT NULL,
+  software_id int unsigned DEFAULT NULL,
+  os_version_id int unsigned DEFAULT NULL,
+  PRIMARY KEY (id),
+  UNIQUE KEY idx_kernels_unique_mapping (software_title_id,software_id,os_version_id)
```

Comment:
```suggestion
  UNIQUE KEY idx_kernels_unique_mapping (os_version_id, software_id)
```

Don't need software title ID in the unique key I think (titles -> software is 1:M), but we _do_ want OS version ID first in the multi-column index because we'll be filtering by OS version ID basically all of the time. This change should make things more performant for both reads and writes.

With that said, I was expecting a differently-shaped table here. I figured that host counts would be the slow part of this query, similar to the reason we have a cron-built host software counts table, and this doesn't include counts at all so we'd need to inventory host counts per kernel manually, still joining against the hosts table etc. This is because you could have two hosts with the same kernel and different operating systems, and we shouldn't show those two hosts in the same count (should be able to write a test for this).